requests_toolbelt/adapters/appengine.py

```diff
@@ -44,6 +49,66 @@ def __init__(self, validate_certificate=True, *args, **kwargs):
         super(AppEngineAdapter, self).__init__(self, *args, **kwargs)
 
     def init_poolmanager(self, connections, maxsize, block=False):
-        self.poolmanager = gaecontrib.AppEngineManager(
-            validate_certificate=self._validate_certificate
-        )
+        self.poolmanager = _AppEnginePoolManager(self._validate_certificate)
+
+
+class _AppEnginePoolManager(object):
+    """Implements urllib3's PoolManager API."""
+
+    def __init__(self, validate_certificate=True):
+        self.validate_certificate = validate_certificate
+
+    def connection_from_url(self, url):
+        return _AppEngineConnection(self, url)
+
+    def clear(self):
+        pass
+
+
+class _AppEngineConnection(object):
+    """Implements urllib3's HTTPConnectionPool API's urlopen().
+
+    This Connection's urlopen() is called with a host-relative path,
+    so in order to properly support opening the URL, we need to store the full URL
+    when we are constructed from the Manager.
+    """
+
+    def __init__(self, pool_manager, url):
+        self.appengine_manager = gaecontrib.AppEngineManager(
```

Comment:
Could you have the `pool_manager` construct this and pass it down to each connection?